cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1004,6 +985,9 @@ VersionedItem LocalVersionedEngine::compact_incomplete_dynamic(
 
     version_map_->write_version(store_, versioned_item.key_);
```

Comment:
Should this not be replaced by the call to `write_version_and_prune_previous_if_needed`? **It was in the previous PR**

Reply:
Oops. Pushed a fix.